src/main/java/com/amazon/dlic/auth/http/jwt/HTTPJwtAuthenticator.java

```diff
@@ -51,9 +53,12 @@
 
     protected final Logger log = LogManager.getLogger(this.getClass());
 
+    private static final Pattern BASIC = Pattern.compile("^\\s*Basic\\s.*", Pattern.CASE_INSENSITIVE);
```

Comment:
Instead of skipping basic Auth headers, can we skip any non "bearer" headers ?

Reply:
It sounds that "Bearer" scheme is optional. Please see `HTTPJwtAuthenticatorTest.testNonBearer()` that test that `HTTPJwtAuthenticator` can accept JWT in `Authorization` header without `Bearer`.

Reply:
I see.

Would be great if you can add a unit-test for this use-case.
In addition to this, there might be other "Authorization" headers that needs to be checked similarly as "Basic".

I would prefer also to discuss this internally if we should push this change, because usually this issue is happening when users configure incorrectly the order of authentication configurations.

Reply:
> In addition to this, there might be other "Authorization" headers that needs to be checked similarly as "Basic".

Can be addressed in a follow up PR if necessary.

> I would prefer also to discuss this internally if we should push this change, because usually this issue is happening when users configure incorrectly the order of authentication configurations.

Actually it is the correct order. Putting Basic authenticator in front will cause `No 'Basic Authorization' header, send 401 and 'WWW-Authenticate Basic'` warning message even when a valid JWT token is present and inability to login with JWT. Please see [HTTPHelper](https://github.com/opendistro-for-elasticsearch/security/blob/6fe4e0ffca29af61afa8492d7afd5846a3e6753a/src/main/java/com/amazon/opendistroforelasticsearch/security/support/HTTPHelper.java#L48)

Reply:
@dinusX how we can set JWT request header token in request payload